### PR TITLE
Multilocale libraries now compile client library with host compiler

### PR DESCRIPTION
**Pull request by dlongnecke-cray**
This PR adjusts multilocale library compilation so that the client library is compiled by the `CHPL_HOST_COMPILER` instead of the `CHPL_TARGET_COMPILER`.

---

Testing:

- [x] `ALL` on `linux64` when `CHPL_COMM=gasnet`, `HOST/TARGET=gnu`
- [x] `interop/C/multilocale` on `darwin` when `CHPL_COMM=gasnet`, `HOST/TARGET=clang`
- [x] `interop/python/multilocale` on `darwin` when `CHPL_COMM=gasnet`, `HOST/TARGET=clang`

For Crays, `interop/C/multilocale/static/use_testing.ml-test.c`, `CHPL_COMM=ugni`:

- [x] `HOST=gnu`, `TARGET=PrgEnv-gnu`
- [x] `HOST=gnu`, `TARGET=PrgEnv-intel`
- [x] `HOST=gnu`, `TARGET=PrgEnv-cray`

### CODE REVIEW

**Comment by dlongnecke-cray** [comment]
@lydia-duncan 

Would you be willing to run the `darwin` tests for me? Something is broken on my local machine and my multilocale libraries are hanging indefinitely.

File: runtime/etc/Makefile.mli-common [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#
+HOST_CC=$(call getHostBaseValue,CC)
+TARG_CC=$(CC)
+
+#
+# These two sets of flags are used when compiling the launcher object files
+# in `Makefile.launcher`. We mimick them here when compiling the client
+# library.
+#
+HOST_CFLAGS=$(call runHostRule,runtime/etc/Makefile.include,printcflags)
```
Comment (ronawho) [inline runtime/etc/Makefile.mli-common]: Personally, I find trying to stuff host and target flags into the same file confusing. Can we just split out the mli makefile into host and target files? 

That way we could just continue to use the normal `CHPL_MAKE_HOST_TARGET = --{host,target}` scheme instead of trying to get both host and target values in one makefile

Comment (gbtitus) [reply runtime/etc/Makefile.mli-common]: > Can we just split out the mli makefile into host and target files?

I think I agree with this.  The coding to handle both host and target in one place is a bit hard to follow, for me.  It appears that if we split into, say:

    Makefile.mli-host-shared
    Makefile.mli-host-static
    Makefile.mli-target-shared
    Makefile.mli-target-static

that we could both get rid of Makefile.mli-common and simplify the coding in the remaining 4 files by using `$(CC)` in place of `$(HOST_CC)` and such.

File: runtime/etc/Makefile.mli-common [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+# host compiler (CHPL_HOST_COMPILER), while the Chapel server (what is
+# launched on the compute nodes) is compiled with the target compiler
+# (CHPL_TARGET_COMPILER).
+#
+HOST_CC=$(call getHostBaseValue,CC)
+TARG_CC=$(CC)
+
+#
+# These two sets of flags are used when compiling the launcher object files
+# in `Makefile.launcher`. We mimick them here when compiling the client
```
Comment (gbtitus) [inline runtime/etc/Makefile.mli-common]: Sp: `mimic`.  (though this may be moot given the comments below)

**Comment by dlongnecke-cray** [comment]
I'm going to close this for now. I'm working on implementing a recursive Make implementation similar to the one suggested by @bradcray. 

It will launch a sub-make that compiles the client as host.
